websites/R/RTBF Auvio/presence.ts

```diff
@@ -0,0 +1,701 @@
+import { ActivityType, Assets, getTimestamps, getTimestampsFromMedia, timestampFromFormat } from 'premid'
+import {
+  ActivityAssets,
+  checkStringLanguage,
+  cropPreset,
+  exist,
+  formatDuration,
+  getChannel,
+  getColor,
+  getLocalizedAssets,
+  getSetting,
+  getThumbnail,
+  limitText,
+  presence,
+  strings,
+} from './util.js'
+
+const browsingTimestamp = Math.floor(Date.now() / 1000)
+const slideshow = new Slideshow()
+
+let oldPath = document.location.pathname
+let isMediaPage = false
+let isMediaPlayer = false
+let title = ''
+let subtitle = ''
+
+presence.on('UpdateData', async () => {
+  const { href, pathname } = document.location
+  const pathParts = pathname.split('/')
+  const presenceData: PresenceData = {
+    name: 'Auvio',
+    largeImageKey: ActivityAssets.Auvio, // Default
+    largeImageText: 'RTBF Auvio',
+    smallImageKey: ActivityAssets.Binoculars,
+    smallImageText: strings.browsing,
+    type: ActivityType.Watching,
+  }
+  const [
+    newLang,
+    usePresenceName,
+    useChannelName,
+    usePrivacyMode,
+    usePoster,
+    useButtons,
+  ] = [
+    getSetting<string>('lang', 'en'),
+    getSetting<boolean>('usePresenceName'),
+    getSetting<boolean>('useChannelName'),
+    getSetting<boolean>('usePrivacyMode'),
+    getSetting<boolean>('usePoster'),
+    getSetting<number>('useButtons'),
+  ]
+
+  // Update strings if user selected another language.
+  if (!checkStringLanguage(newLang))
+    return
+
+  if (oldPath !== pathname) {
+    oldPath = pathname
+    slideshow.deleteAllSlides()
+  }
+
+  let useSlideshow = false
+
+  switch (true) {
+    case exist('#audioPlayerContainer') && document.querySelector('#PlayerUIAudioPlayPauseButton')?.getAttribute('aria-label') === 'pause': {
+      /* NOTE: PODCAST PLAYER
+      NOTE: When a podcast is played, it appears in an audio player at the bottom of the screen.
+      Once a podcast has been launched, it is visible at all times throughout the site until the website is refreshed. */
+
+      const firstLine = document.querySelector('.PlayerUIAudio_titleText__HV4Y2')?.textContent ?? ''
+      const secondLine = document.querySelector('.PlayerUIAudio_subtitle__uhGA4')?.textContent ?? ''
+      const duration = document.querySelector('.PlayerUIAudio_duration__n7hxV')?.textContent ?? '0'
```

Comment:
If possible, try using selectors which exclude generated class names. (applies to other locations in the code as well)
```ts
      const firstLine = document.querySelector('[class*=PlayerUIAudio_titleText]')?.textContent ?? ''
      const secondLine = document.querySelector('[class*=PlayerUIAudio_subtitle]')?.textContent ?? ''
      const duration = document.querySelector('[class*=PlayerUIAudio_duration]')?.textContent ?? '0'
```

Reply:
Thanks for the review, I'll look into it asap! I didn't know about these selectors